core/app/models/spree/payment_method.rb

```diff
@@ -7,7 +7,7 @@ class PaymentMethod < Spree::Base
 
     scope :active,                 -> { where(active: true) }
     scope :available,              -> { active.where(display_on: [:front_end, :back_end, :both]) }
-    scope :available_on_front_end, -> { active.where(display_on: [:front_end, :both]) }
+    scope :available_on_front_end, -> { active.where(display_on: [:front_end, :both]).order(position: :asc) }
```

Comment:
@krtschmr  Its good if we append ```order(position: :asc)```  for active scope, which internally used in rest of scopes.
 @damianlegawiec What's your thoughts here?